alibi_detect/od/_pca.py

```diff
@@ -0,0 +1,184 @@
+from typing import Union, Optional, Callable, Dict, Any
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from alibi_detect.utils._types import Literal
+from alibi_detect.base import outlier_prediction_dict
+from alibi_detect.base import BaseDetector, ThresholdMixin, FitMixin
+from alibi_detect.od.pytorch import KernelPCATorch, LinearPCATorch
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+from alibi_detect.exceptions import _catch_error as catch_error
+
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': (KernelPCATorch, LinearPCATorch)
+}
+
+
+class PCA(BaseDetector, ThresholdMixin, FitMixin):
+    def __init__(
+        self,
+        n_components: int,
+        kernel: Optional[Callable] = None,
+        backend: Literal['pytorch'] = 'pytorch',
```

Comment:
Is the itention also to provide a `keops` backend in the future? Like with Mahalonobis...

Reply:
Yep